web/src/lib/components/asset-viewer/video-remote-viewer.svelte

```diff
@@ -0,0 +1,95 @@
+<script lang="ts">
+  import CircleIconButton from '$lib/components/elements/buttons/circle-icon-button.svelte';
+  import Icon from '$lib/components/elements/icon.svelte';
+  import LoadingSpinner from '$lib/components/shared-components/loading-spinner.svelte';
+  import { castManager, CastState } from '$lib/managers/cast-manager.svelte';
+  import { mdiCastConnected, mdiPause, mdiPlay } from '@mdi/js';
+  import { t } from 'svelte-i18n';
+
+  interface Props {
+    poster: string;
+    assetFileUrl: string;
+    onVideoStarted: () => void;
+    onVideoEnded: () => void;
+  }
+
+  let { poster, assetFileUrl, onVideoEnded, onVideoStarted }: Props = $props();
+
+  let previousPlayerState: CastState | null = $state(null);
+
+  const handlePlayPauseButton = () => {
+    if (castManager.castState === CastState.PLAYING) {
+      castManager.pause();
+    } else if (castManager.castState === CastState.IDLE) {
+      void cast(assetFileUrl, true);
+    } else {
+      castManager.play();
+    }
+  };
+
+  $effect(() => {
+    if (assetFileUrl) {
+      void cast(assetFileUrl);
+    }
+  });
+
+  $effect(() => {
+    if (
+      castManager.castState !== previousPlayerState &&
```

Comment:
This case is impossible, right? This effect will only trigger *if* `castState` (or `previousPlayerState` I guess) has changed, in which case it isn't the previous state anymore.